### PR TITLE
Build release & sign only on merge / release / release PR

### LINKED ISSUE #982: Decouple tauri build/sign in CI
I really don’t love that we have to wait for that super long CI for every commit. Maybe we could run a first simplified Linux tauri build just for testing purposes, with e2e and seing if tauri build works. And build all platforms + sign only on merge to main, release PRs, and releases.

Would like to keep tauri in the loop for every commit and PR, but keep that simple and quick (no need for secrets for external contributors for instance)

### PR DESCRIPTION
**Pull request by pierremtb**
Fixes #982 plus slight reorg, see comments

image

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -46,6 +50,20 @@ jobs:
       - run: yarn tsc
 
 
+  check-typos:
```
Comment (pierremtb) [inline .github/workflows/ci.yml:53]: Moved up from later in the file for consistency

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -102,7 +120,7 @@ jobs:
 
 
   build-test-apps:
-    needs: [check-format, build-test-web, prepare-json-files, check-types]
+    needs: [prepare-json-files]
```
Comment (pierremtb) [inline .github/workflows/ci.yml:123]: Reduce deps here, those can run in parallel. publish-apps now depends on all of these tho

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -165,13 +183,13 @@ jobs:
       - name: Fix format
         run: yarn fmt
 
-      - name: install apple silicon target mac
+      - name: Install Universal target (macos only)
```
Comment (adamchalmers) [inline .github/workflows/ci.yml:186]:
```suggestion
      - name: Install Universal target (MacOS only)
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -165,13 +183,13 @@ jobs:
...
         if: matrix.os == 'macos-latest'
         run: |
           rustup target add aarch64-apple-darwin
 
-      - name: Prepare Windows certificate and variables
-        if: matrix.os == 'windows-latest'
+      - name: Prepare certificate and variables (windows only)
```
Comment (adamchalmers) [inline .github/workflows/ci.yml:191]:
```suggestion
      - name: Prepare certificate and variables (Windows only)
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -185,8 +203,8 @@ jobs:
           echo "C:\Program Files\DigiCert\DigiCert One Signing Manager Tools" >> $GITHUB_PATH
         shell: bash
 
-      - name: Setup Windows certicate with SSM KSP
-        if: matrix.os == 'windows-latest'
+      - name: Setup certicate with SSM KSP (windows only)
```
Comment (adamchalmers) [inline .github/workflows/ci.yml:206]:
```suggestion
      - name: Setup certicate with SSM KSP (Windows only)
```

**Comment by adamchalmers** [comment]
Nice!

**Pull request merged**